File: UTESA/DCE 5 Estrellas/dce2005/Estrella1/DCE1_Ejemplos/src/VB/Cliente/Opcion.vb
''' <summary>
''' Representa una opción genérica.
''' Todas las opciones del sistema deben heredar de esta opción.
''' </summary>
Public MustInherit Class Opcion

    Protected _Codigo As Integer = 0
    Protected _Descripcion As String

    ''' <summary>
    ''' El código de la opción.
    ''' </summary>
    Protected Friend Property Codigo() As Integer
        Get
            Return _Codigo
        End Get
        Set(ByVal Value As Integer)
            _Codigo = Value
        End Set
    End Property

    ''' <summary>
    ''' La descripción de la opción.
    ''' </summary>
    Protected Friend Property Descripcion() As String
        Get
            Return _Descripcion
        End Get
        Protected Set(ByVal Value As String)
            _Descripcion = Value
        End Set
    End Property

    ''' <summary>
    ''' Ejecuta la acción asociada a la opción.
    ''' </summary>
    Friend MustOverride Sub EjecutarAccion()

End Class

File: UTESA/DCE 5 Estrellas/dce2005/Estrella1/DCE1_Ejemplos/src/CS/Cliente/Opcion.cs
using System;
using System.Collections.Generic;
using System.Text;

using DCE05.Ejemplos.EstrellaUno.ReglasNegocio;

namespace DCE05.Ejemplos.EstrellaUno.Cliente {
    
    /// <summary>
    /// Representa una opción genérica.
    /// Todas las opciones del sistema deben heredar de esta opción.
    /// </summary>
    internal abstract class Opcion {

        private int codigo = 0;
        private string descripcion;

        /// <summary>
        /// El código de la opción.
        /// </summary>
        protected internal int Codigo {
            get { return codigo; }
            set { codigo = value; }
        }

        /// <summary>
        /// La descripción de la opción.
        /// </summary>
        protected internal string Descripcion {
            get { return descripcion; }
            protected set { descripcion = value; }
        }

        /// <summary>
        /// Ejecuta la acción asociada a la opción.
        /// </summary>
        internal abstract void EjecutarAccion();

    }
}
